ugvc/filtering/multiallelics.py

```diff
@@ -0,0 +1,526 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+import pyfaidx
+import pysam
+
+import ugvc.comparison.flow_based_concordance as fbc
+import ugvc.filtering.training_prep as tprep
+import ugvc.flow_format.flow_based_read as fbr
+from ugvc.filtering.tprep_constants import SPAN_DEL
+from ugvc.vcfbed import vcftools
+
+
+def select_overlapping_variants(df: pd.DataFrame) -> list:
+    """Selects lists of overlapping variants that need to be genotyped together. This
+    can be multiallelic variants or variants with spanning deletion
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Training set dataframe
+    Returns
+    -------
+    list
+        List of list of indices that generate the co-genotyped sets
+    """
+    multiallelic_locations = set(np.where(df["alleles"].apply(len) > 2)[0])
+    del_length = df["alleles"].apply(lambda x: max(len(x[0]) - len(y) for y in x))
+    current_span = 0
+    results = []
+    cluster = []
+    for i in range(df.shape[0]):
+        if len(cluster) == 0 and del_length[i] == 0:
+            continue
+        if df.iloc[i]["pos"] > current_span:
+            if len(cluster) > 1:
+                for c in cluster:
+                    if c in multiallelic_locations:
+                        multiallelic_locations.remove(c)
+                results.append(cluster[:])
+            cluster = []
+        if len(cluster) == 0:
+            cluster.append(i)
+        elif len(cluster) > 0 and SPAN_DEL in df.iloc[i]["alleles"]:
```

Comment:
If there is no spanning deletion in the variant, then it is not genotyped together with the potentially spanning deletion. I am not 100% how this can happen. I added comments in this for loop that hopefully clarify